[ElonKoh] iP

ElonKoh opened the pull request:
Level-1
Level-2
Level-3
A-CodingStandard

Commit:
Renamed Ebot and added Greeting and Exit message

ElonKoh committed:
Added echo-user functionality and some chat formatting

ElonKoh committed:
Added TaskList Class and the ability to add tasks to a TaskList

ElonKoh committed:
Added Task Class and ability to mark and unmark Tasks

ElonKoh committed:
Abstracted 1) Final print attributes, 2) TaskList reply and mark/unmark methods, and 3) Added comments for readability

ElonKoh committed:
Added Task subclasses - Todo, Event, Deadline

ElonKoh committed:
Override toString methods for task subclass

wentuoc commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+                System.out.println(lineDivider);
+
+            // if user types "list" command
+            } else if(userInput.equalsIgnoreCase("list")) {
```
Can consider adding a space between `if` and the parentheses
```suggestion
            } else if (userInput.equalsIgnoreCase("list")) {
```

wentuoc commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+                System.out.println(bigIndent + "[X] " + taskToMark.getDescription());
+                System.out.println(lineDivider);
+            }
+        } catch(NullPointerException E) {
```
Can consider adding a space after `catch`

wentuoc commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,63 @@
+        numberOfTaskLists += 1;
+    }
+
+    public void AddEntry(Task entry) {
```
Can consider changing to camel case for `addEntry`

wentuoc commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,63 @@
+    public String toString() {
+        String printList = "";
+        for (int i = 0; i < numberOfEntries; i++) {
+            String printEntry = (i+1) + ". " + entries[i] + System.lineSeparator();
```
Can consider adding spaces in `(i + 1)`

wentuoc commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    private static void handleMarkingOutput(String userInput, TaskList taskList, String unmarkCommand, String markCommand) {
+        if(userInput.toLowerCase().contains(unmarkCommand)) {
```
Can consider adding a space after `if`

wentuoc reviewed (commented):
Overall good adherence to coding standards! Can consider adding some spaces after `if`.

shira421 commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+            taskList.AddEntry(newTask);
+            System.out.println(smallIndent + "Got it! I added a new to-do: " + System.lineSeparator() + bigIndent + newTask);
+            System.out.println(lineDivider);
+        } else if(userInput.toLowerCase().contains("event")) {
```
do add a space after `else if`

shira421 commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+        }
+    }
+
+    private static void handleListOutput(TaskList taskList) {
```
good use of Camel Case!

shira421 commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,156 @@
+        int unmarkIndex = userInput.indexOf(unmarkString);
+        int taskNumPointer = (unmarkIndex + unmarkString.length());
+        int taskNum = Character.getNumericValue(userInput.charAt(taskNumPointer));
+        try {
+            Task taskToUnmark = taskList.getTask(taskNum - 1);
+            if (!taskToUnmark.isDone()) {
+                System.out.println(bigIndent + "Task " + taskNum + " is already marked as not done");
+                System.out.println(bigIndent + "[ ] " + taskToUnmark.getDescription());
+                System.out.println(lineDivider);
+            } else {
+                taskToUnmark.setIsDone(false);
+                System.out.println(bigIndent + "Got it! I marked task " + taskNum + " as not done");
+                System.out.println(bigIndent + "[ ] " + taskToUnmark.getDescription());
+                System.out.println(lineDivider);
+            }
+        } catch(NullPointerException E) {
+            System.out.println(bigIndent + "Task not found");
+            System.out.println(lineDivider);
+        }
```
I like the use of try-catch here.

shira421 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Override
+    public String toString() {
+        String taskCheck;
+        if (super.isDone()) {
+            taskCheck = "X";
+        } else {
+            taskCheck = " ";
+        }
```
perhaps this can be compressed by initialising taskCheck as " " and changing it to "X" if `super.isDone()` is true

ElonKoh committed:
Fixed coding standards to adhere to naming conventions, and simplify code to be more legible

ElonKoh committed:
Added error handling for empty descriptions, missing keywords, marking tasks that are already marked

ElonKoh committed:
Added error handling exceptions for missing information

ElonKoh committed:
Merge branch 'Branch-Level-5'

* Branch-Level-5:
  Added error handling exceptions for missing information
  Added error handling for empty descriptions, missing keywords, marking tasks that are already marked

ElonKoh committed:
Added ability to delete tasks in TaskList by switching to ArrayList structure

ElonKoh committed:
Added creation of output file to save task list to ./data/ebot.txt

ElonKoh committed:
Merge branch 'branch-level-6'

* branch-level-6:
  Added ability to delete tasks in TaskList by switching to ArrayList structure

ElonKoh committed:
Merge branch 'branch-level-7'

* branch-level-7:
  Added creation of output file to save task list to ./data/ebot.txt

# Conflicts:
#	src/main/java/Ebot.java

ElonKoh committed:
Resolved merge conflicts

ElonKoh committed:
Fixed merge conflict issues and print formatting

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+import java.io.FileWriter;
+
+public class Ebot {
+    public static final String OUTPUT_FILE_NAME = "./data/ebot.txt";
+    private static final String LINE_DIVIDER = ("___________________________________________");
+    private static final String BIG_INDENT = "\t\t";
+    private static final String SMALL_INDENT = "\t";
+
```
Good job with using constants for this

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+    private static final String BIG_INDENT = "\t\t";
+    private static final String SMALL_INDENT = "\t";
+
+    public static void main(String[] args) {
```
This function >30 LOC consider refactoring the inner lines into other methods

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+                SMALL_INDENT + "What can i do for you?");
+        String exitMessage = (SMALL_INDENT + "Bye. Hope to see you again soon!");
+
+        // Program start, initialize scanner for user inputs
+        String userInput;
+        Scanner in = new Scanner(System.in);
+
+        // Initialize list
+        int numberOfLists = TaskList.getNumberOfTaskLists();
+        String taskListName = ("list") + numberOfLists;
+        TaskList taskList = new TaskList(taskListName);
+
+        // Initialize file
+        createFile();
+
+        // Print Welcome Message
+        System.out.println(LINE_DIVIDER + System.lineSeparator() + welcomeMessage);
+        System.out.println(LINE_DIVIDER);
+
+        // User chat loop
```
You don't need to write comments for certain snippets if the code is already clear

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+            taskList.deleteEntry(deleteIndex);
+            System.out.println(SMALL_INDENT + "Got it! I deleted task " + (deleteIndex + 1));
+            System.out.println(taskList);
+        } catch (NumberFormatException E) {
+            System.out.println(SMALL_INDENT + "Error! Did you forget to indicate which index to delete?");
+        } catch (IndexOutOfBoundsException E) {
```
Variable names (here, the exception object `E`) should be in `pascalCase`

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+        String cleanedUserInput = userInput.toLowerCase().replace(" ", "").replace("/", "");
+        String taskDescription;
+        Task newTask;
+        if (userInput.toLowerCase().contains("todo")) {
+            taskDescription = cleanedUserInput.replace(("todo"), "");
+            try {
+                if (taskDescription.length() == 0) {
+                    throw new MissingDescriptionException();
+                }
+                newTask = new Todo(taskDescription, false);
```
this is 3 levels of indentation, consider refactoring inner lines of code to another method

sevenseasofbri commented on src/main/java/Ebot.java:
```diff
@@ -0,0 +1,256 @@
+        }
+    }
+
+    private static void handleTaskAdding(TaskList taskList, String userInput) {
```
this function is also very long, you can break it up into smaller functions to improve readability

sevenseasofbri reviewed (commented):
Good job so far! Some functions are a little long, but otherwise most are well written.

ElonKoh committed:
ArrayList Implemented for A-Collections

ElonKoh committed:
Built Jar file for Ebot - A-Jar

ElonKoh committed:
Added Parser, Ui, and Storage class. Complete overhaul of code to abstract out more OOP.

ElonKoh committed:
Added Find command to search for tasks

ElonKoh committed:
Added JavaDoc comments for non-private classes/methods

ElonKoh committed:
Merge pull request #1 from ElonKoh/branch-Level-9

Added Find command to search for tasks

ElonKoh committed:
Merge pull request #2 from ElonKoh/branch-A-JavaDoc

Added JavaDoc comments for non-private classes/methods

ElonKoh committed:
Merge commit 'dea1fc9cef4f4bf2c9d6c467866f08c2b112cb1e'

* commit 'dea1fc9cef4f4bf2c9d6c467866f08c2b112cb1e':
  Added JavaDoc comments for non-private classes/methods
  Added Find command to search for tasks

ElonKoh committed:
Updated ReadMe file

ElonKoh committed:
Updated docs/README

ElonKoh committed:
Fixed hardcoded paths for JAR release, includes Ebot 2.0 as a jar release

ElonKoh committed:
Fixed FileWriter to save changes constantly in real-time

ElonKoh committed:
Re-release for final release

ElonKoh committed:
Fixed index out of bounds error when "mark 0" is input
Fixed commands to accept /from /to /by instead
Implemented loading tasks from saved txt file

ElonKoh committed:
Fixed README files to accommodate change in command keywords to include "/"

ElonKoh committed:
Finalised bug-tested JAR release
